src/_pytest/python.py

```diff
@@ -1096,6 +1097,37 @@ def _validate_if_using_arg_names(self, argnames, indirect):
                         pytrace=False,
                     )
 
+    def _validate_explicit_parameters(self, argnames, indirect):
+        """
+        The argnames in *parametrize* should either be declared explicitly via
+        indirect list or explicitly in the function
+
+        :param List[str] argnames: list of argument names passed to ``parametrize()``.
+        :param indirect: same ``indirect`` parameter of ``parametrize()``.
+        :raise ValueError: if validation fails
+        """
+        func_name = self.function.__name__
+        if type(indirect) is bool and indirect is True:
+            return
+        temp_argnames = list()
```

Comment:
`temp_argnames` is not a very good name in general, I suggest `parametrized_argnames` instead.